src/coreclr/jit/gentree.cpp

```diff
@@ -20893,11 +20894,34 @@ GenTree* Compiler::gtNewSimdBinOpNode(
 
     if (intrinsic != NI_Illegal)
     {
+        if (op == GT_AND_NOT)
+        {
+            assert(fgNodeThreading == NodeThreading::LIR);
+
+#if defined(TARGET_XARCH)
+            // GT_AND_NOT expects `op1 & ~op2`, but xarch does `~op1 & op2`
+            // We specially handle this here since we're only producing a
+            // native intrinsic node in LIR
+
+            std::swap(op1, op2);
```

Comment:
can you remind me - is it fine to do this `swap` here in terms of side-effect reordering?

Reply:
ah, it's LIR so I guess it is

Reply:
Right, in this case its fine specifically because we've validated we're in LIR already.

Otherwise, we should be applying `GTF_REVERSE_OPS` or have some comment about expecting the user to have already spilled side effects, etc.